File: solutions/golang/socialnetworkingservice/notification.go
package socialnetworkingservice

import "time"

type Notification struct {
	ID        string
	UserID    string
	Type      NotificationType
	Content   string
	Timestamp time.Time
}

func NewNotification(id, userID string, notifType NotificationType, content string) *Notification {
	return &Notification{
		ID:        id,
		UserID:    userID,
		Type:      notifType,
		Content:   content,
		Timestamp: time.Now(),
	}
}

File: solutions/csharp/socialnetworkingservice/Notification.cs
using System;

namespace SocialNetworkingService
{
    public class Notification
    {
        public string Id { get; }
        public string UserId { get; }
        public NotificationType Type { get; }
        public string Content { get; }
        public DateTime Timestamp { get; }

        public Notification(string id, string userId, NotificationType type, string content, DateTime timestamp)
        {
            Id = id;
            UserId = userId;
            Type = type;
            Content = content;
            Timestamp = timestamp;
        }
    }
}